Assets/World/Behavior/Selectable/Selectable.gd

```
@@ -0,0 +1,59 @@
+extends Node
+##
+
+class_name Selectable
+
+## The sprite to be highlighted when selected.
+## If null, will use the first found animated sprite sibling.
+@export var sprite: Node2D
+## The shader to be used to highlight the sprite.
+@export var shader: ShaderMaterial = preload("res://Assets/World/Behavior/Selectable/SelectableDefultShader.tres")
+
+@onready var parent: Node2D = self.get_parent()
+
+## The shader`s defult width
+var shader_width: float = shader.get_shader_parameter("width")
+## Whether the object is currently selected
+var is_selected: bool = false:
+  get:
+    return is_selected
+  set(value):
+    is_selected = value
+    # change the shader transparency to the correct one.
+    if sprite:
+      if is_selected:
+        sprite.material.set_shader_parameter("width", shader_width)
+      else:
+        sprite.material.set_shader_parameter("width", 0)
+
+func _ready():
+  if not sprite or not (sprite is AnimatedSprite2D or sprite is Sprite2D):
```

fail fast: assert at the moment sprite is set